packages/cli/index.ts

```diff
@@ -311,23 +308,32 @@ function hijackDeps(
 }
 
 async function verifyCompactMode(packageName: string) {
-  const error = new Error(
-    `pkg-pr-new cannot resolve ${packageName} from npm. --compact flag depends on the package being available in npm.
-Make sure to have your package on npm first or configure the 'repository' field in your package.json properly.`,
-  );
+  let manifest: PackageManifest;
+
   try {
-    const manifest = await getPackageManifest(packageName);
+    manifest = await getPackageManifest(packageName);
+  } catch {
+    throw new Error(
+      `pkg-pr-new cannot resolve ${packageName} from npm. --compact flag depends on the package being available in npm.
+Make sure to have your package on npm first.`,
+    );
+  }
 
-    const repository = extractRepository(manifest);
-    if (!repository) {
-      throw error;
-    }
+  const repository = extractRepository(manifest);
+  if (!repository) {
+    throw new Error(
+      `pkg-pr-new cannot extract the repository link from the ${packageName} manifest. --compact flag requires the link to be present.
+Make sure to configure the 'repository' / 'repository.url' field in its package.json properly.
+See https://docs.npmjs.com/cli/v10/configuring-npm/package-json#repository for details.`,
+    );
+  }
 
-    const match = extractOwnerAndRepo(repository);
-    if (!match) {
-      throw error;
-    }
-  } catch {
-    throw error;
+  const match = extractOwnerAndRepo(repository);
+  if (!match) {
+    throw new Error(
+      `pkg-pr-new cannot extract the owner and repo names from the ${packageName} repository link: ${repository}. --compact flag requires these names.
+Make sure to configure the 'repository' / 'repository.url' field in its package.json properly.
```

Comment:
can we make this part that is present in the two error messages a static variable or something like that? so we reuse it! 

like `const instruction` in the top of the function

Reply:
Addressed in `fd461c3`.